elf/elf.go

```diff
@@ -696,9 +717,9 @@ func (b *Module) Load() error {
 			}
 
 			// assign perf fd tp map
-			ret := C.bpf_update_element(C.int(b.maps[name].m.fd), unsafe.Pointer(&cpu), unsafe.Pointer(&pmuFD), C.BPF_ANY)
-			if ret != 0 {
-				return fmt.Errorf("cannot assign perf fd to map: %d (cpu %d)", syscall.Errno(ret), cpu)
+			_, err = C.bpf_update_element(C.int(b.maps[name].m.fd), unsafe.Pointer(&cpu), unsafe.Pointer(&pmuFD), C.BPF_ANY)
```

Comment:
I see the cgo magic makes the function `C.bpf_update_element()` both the return value of `bpf()` system call and errno.

However, in case of success, `errno` might not be reset to zero so it might keep a non-zero value from a previous unrelated function.

man 3 errno:
> Its value is significant only when the return value of the call indicated an error

"man 2 bpf" does not promise to set errno to zero in case of success.

So I think the error check below is not correct.